src/dali/plugin/pair_converter/csv/kraken.py

```diff
@@ -365,23 +390,23 @@ def _retrieve_cached_bars(
                     with gopen(file_path, "rt") as file:
                         rows = reader(file)
                         for row in rows:
-                            if all_bars and int(row[self.__TIMESTAMP_INDEX]) >= duration_timestamp:
+                            if all_bars and int(float(row[self.__TIMESTAMP_INDEX])) >= duration_timestamp:
```

Comment:
Why the double cast (float + int)?

Reply:
One of my files had a decimal point in the timestamp for some reason and I was getting an error with int. Is there a better way?

Reply:
Does casting directly to int work?

Reply:
nevermind, I found the kink in the line. When emulating the 1 week candle, the timestamp that is returned in line 312 is a float:
https://github.com/eprbell/dali-rp2/blob/508dc43cbfc0bd99dfef85c522f3cecb6055df9d/src/dali/plugin/pair_converter/csv/kraken.py#L308
I cast it as an int, then a str. I'm running real world data now to see if that fixes the issue.

Reply:
The casting fixed it. I guess I got a little lazy there.